src/Products/PageTemplates/engine.py

```diff
@@ -135,6 +136,13 @@ def _compile_zt_expr(type, expression, engine=None, econtext=None):
     """
     if engine is None:
         engine = econtext["__zt_engine__"]
+    # *expression* is a ``chameleon.tokenize.Token`` when
+    # the template is compiled but "text" when the template code
+    # comes from the ``chameleon`` cache.
+    # Under PY3, ``chameleon`` wrongly translates ``Token``;
```

Comment:
I do think this comment is rather misleading given the nature of the integration into Chameleon's expression translation machinery. A rather substantial explanation of the issue has been provided to explain why a `Token` is provided in the initial compilation pass, and not during rendering.